### PR TITLE
DEV-2025 fix travis and gitlab pipelines

### PR DESCRIPTION
**Pull request by qiaouchicago**

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -54,6 +54,36 @@ tox:
     - !reference [.load_github_key, script]
     - tox -e py
 
+# used by release
+.python_versions:
+  parallel:
+    matrix:
+      - BUILD_PY_VERSION: [python3.6]
+      - RELEASE_PY_VERSION: [python3.6]
```
Comment (wrkhenddher) [inline .gitlab-ci.yml:62]: Shouldn't we add more versions? (... 3.8, 3.9+?)
Doesn't it have to match the builder below?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -54,6 +54,36 @@ tox:
...
+
+release:
+  before_script:
+    - git fetch --unshallow
+
+get_version:
+  stage: test
+  tags:
+    - dind
+  image: ${BASE_CONTAINER_REGISTRY}/python3.8-builder:${BASE_CONTAINER_VERSION}
```
Comment (wrkhenddher) [inline .gitlab-ci.yml:72]: builder 3.8?

**Review by mpsolano — APPROVED** [review]
Looks good!

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -54,6 +54,36 @@ tox:
...
+  before_script:
+    - git fetch --unshallow
+  script:
+    - pip install importlib-metadata
+    - pip install --no-deps .
```
Comment (wrkhenddher) [inline .gitlab-ci.yml:77]: Nice!!!

Hopefully, soon we'll do that for running tests too: `pip install .[dev]" in tox.

And even later on, maybe even for docker build, assuming the artifact was built and successfully pushed to nexus, `pip install indexd==version`.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -54,6 +54,36 @@ tox:
...
+    - >
+      python -c "from importlib_metadata import version; print(version('indexd'))"
+      > VERSION.txt
+    - cat VERSION.txt
```
Comment (wrkhenddher) [inline .gitlab-ci.yml:81]: Not familiar with this ... will this VERSION.txt file be used for?

File: setup.py [hunk 5]
```diff
@@ -2,11 +2,6 @@
 
 setup(
     name="indexd",
```
Comment (wrkhenddher) [inline setup.py:4]: So we are not using setuptools_scm?
Wouldn't that just leave us maintaining our own "version" generation piece?
I know you mentioned that setuptools_scm didn't quite get us there but maybe we are still just missing an important piece of understanding so setuptools_scm does the thing it's designed for?

File: setup.py [hunk 6]
```diff
@@ -2,11 +2,6 @@
...
-    use_scm_version={
-        "local_scheme": "dirty-tag",
-        "write_to": "indexd/_version.py",
```
Comment (wrkhenddher) [inline setup.py:7]: bummer 😢

**Review by wrkhenddher — APPROVED** [review]

File: docker.sh [hunk 7]
```diff
@@ -21,9 +20,15 @@ fi
...
 GIT_BRANCH=${GIT_BRANCH/\//_}
 
+VERSION=$(cat VERSION.txt)
 echo "$VERSION"
 
-docker build --build-arg version="$VERSION" --ssh default -t "$IMAGE_NAME:$GIT_BRANCH" .
+docker build \
+  --label org.opencontainers.image.version="${VERSION}" \
+  --label org.opencontainers.image.created="$(date -Iseconds)" \
+  --label org.opencontainers.image.revision="${CI_COMMIT_SHORT_SHA}" \
```
Comment (kulgan) [inline docker.sh:29]: I think this value is only set for gitlab ci, did you mean to use `${COMMIT}`

File: .gitlab-ci.yml (hunk 1 above)
Comment (qiaouchicago) [reply .gitlab-ci.yml:62]: If I understand correctly, this build version could be used by python3.6+

File: .gitlab-ci.yml (hunk 2 above)
Comment (qiaouchicago) [reply .gitlab-ci.yml:72]: This is just to get the version of indexd. Builder  version should not matter for the result.

File: .gitlab-ci.yml (hunk 4 above)
Comment (qiaouchicago) [reply .gitlab-ci.yml:81]: to provide version for label in docker image.

File: setup.py (hunk 5 above)
Comment (qiaouchicago) [reply setup.py:4]: we are still using setuptools_scm

File: docker.sh (hunk 7 above)
Comment (qiaouchicago) [reply docker.sh]: changed to use git to get commit hash. So it will be independent of the pipeline.

**Pull request merged**